test/sanity/validate-modules/main.py

```diff
@@ -1394,12 +1394,14 @@ def _check_for_new_args(self, doc, metadata):
                     if existing_version:
                         break
                 current_version = details.get('version_added')
-                if current_version != existing_version:
+
+                # We expect the added_version to be a string, not a float
+                if current_version and existing_version and current_version != str(existing_version):
                     self.reporter.error(
                         path=self.object_path,
                         code=309,
                         msg=('version_added for new option (%s) should '
-                             'be %r. Currently %r' %
+                             'be \'%r\'. Currently %r' %
```

Comment:
Please do not quote `%r` as it will automatically add quotes in creating a string representation of a string.

Reply:
Fixed.